mitodl/mitxpro
# Automate environment variables

#### Pre-Flight checklist


#### What are the relevant tickets?
N/A

#### What's this PR do?
This PR adds automation and some guardrails around environment variables. Specifically it does the following:

- `app.json` is generated from `settings.py` definitions
- A test to ensure `settings.py` and `app.json` are in sync
- if a var is required in `settings.py`, it is required in `app.json` and the app will also refuse to start (so required configuration should be in CI/RC/Prod before they are merged into those envs!)
- Environment vars can't be used multiple times


#### How should this be manually tested?
- Add a settings.py var, run `./manage.py generate_app_json` and verify the new var is defined in `app.json`.
- Verify the app will error if a required var is absent
- Look at the `app.json` in this PR and verify the changes make sense and nothing was lost
